File: Semestrul 5/Proiectarea-Detaliata-a-Sistemelor-Software/Laborator/Daria/Laborator_11/command/RemoteController.java
package Laborator.Daria.Laborator_11.command;

public class RemoteController {
    private Command[] onCommands;
    private Command[] offCommands;

    public RemoteController() {
        onCommands = new Command[3];
        offCommands = new Command[3];
    }

    public void setCommand(int slot, Command onCommand, Command offCommand) {
        if (slot >= 0 && slot < 3) {
            onCommands[slot] = onCommand;
            offCommands[slot] = offCommand;
        } else {
            System.out.println("Invalid slot: " + slot);
        }
    }

    public void pushOnButton(int slot) {
        if (slot >= 0 && slot < 3 && onCommands[slot] != null) {
            onCommands[slot].execute();
        } else {
            System.out.println("No command assigned to ON button " + slot);
        }
    }

    public void pushOffButton(int slot) {
        if (slot >= 0 && slot < 3 && offCommands[slot] != null) {
            offCommands[slot].execute();
        } else {
            System.out.println("No command assigned to OFF button " + slot);
        }
    }
}

File: Semestrul 5/Proiectarea-Detaliata-a-Sistemelor-Software/Laborator/Paul/design-patterns/command/src/main/kotlin/RemoteController.kt
package org.paul

class RemoteController {
    private val onCommands: MutableMap<Int, Command> = mutableMapOf()
    private val offCommands: MutableMap<Int, Command> = mutableMapOf()

    fun setCommand(slot: Int, onCommand: Command, offCommand: Command) {
        onCommands[slot] = onCommand
        offCommands[slot] = offCommand
    }

    fun pressOnButton(slot: Int) {
        onCommands[slot]?.execute()
    }

    fun pressOffButton(slot: Int) {
        offCommands[slot]?.execute()
    }
}